environments/Dockerfile

```
@@ -15,32 +15,36 @@ ENV DEBIAN_FRONTEND="noninteractive" \
     LANG="C.UTF-8" \
     PYTHONPATH=${APPLICATION_DIRECTORY}
 
-# Following is needed to install python 3.7
-RUN apt update && apt install --no-install-recommends -y software-properties-common
-RUN add-apt-repository ppa:deadsnakes/ppa
-
-RUN apt update && apt install --no-install-recommends -y \
-    git curl make ssh openssh-client \
-    python${PYTHON_VERSION} python3-pip python-is-python3
-
-# Following is needed to swtich default python3 version
-# For detail, please check following link https://unix.stackexchange.com/questions/410579/change-the-python3-default-version-in-ubuntu
-RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python${PYTHON_VERSION} 1 \
-    && update-alternatives --set python3 /usr/bin/python${PYTHON_VERSION} \
-    # `requests` needs to be upgraded to avoid RequestsDependencyWarning
-    # ref: https://stackoverflow.com/questions/56155627/requestsdependencywarning-urllib3-1-25-2-or-chardet-3-0-4-doesnt-match-a-s
-    && python3 -m pip install --upgrade pip setuptools requests \
+RUN apt update -y && apt install --no-install-recommends -y software-properties-common \
+    # Add deadsnakes PPA to install older Python versions
+    && add-apt-repository ppa:deadsnakes/ppa \
+    && apt-get update && apt-get install -y \
+    # Install specified Python
+    python${PYTHON_VERSION} python${PYTHON_VERSION}-dev python3-pip python3-dev python-is-python3 \
+    build-essential openssl libssl-dev libffi-dev ca-certificates git curl make ssh openssh-client \
+    && update-ca-certificates \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Swtich default python3 version
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python${PYTHON_VERSION} 1 \
+    && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python${PYTHON_VERSION} 1
+
+# Install pip for specified Python version
+RUN curl -sS https://bootstrap.pypa.io/get-pip.py | python${PYTHON_VERSION}
+
+RUN python3 -m pip install --upgrade pip setuptools requests wheel six urllib3 \
     && python3 -m pip install poetry
 
 # Add user. Without this, following process is executed as admin.
 RUN groupadd -g ${GID} ${GROUP_NAME} \
-    && useradd -ms /bin/sh -u ${UID} -g ${GID} ${USER_NAME}
+&& useradd -ms /bin/sh -u ${UID} -g ${GID} ${USER_NAME}
```

It is more consistent with other lines.
```suggestion
    && useradd -ms /bin/sh -u ${UID} -g ${GID} ${USER_NAME}
```